src/core/regimes.rkt

```diff
@@ -76,22 +77,33 @@
   (define prog-body (program-body prog))
   (for/list ([expr (remove-duplicates (subexprs-in-expr prog-body))]
              #:when (and (not (null? (free-variables expr)))
+                         (equal? (type-of expr (for/list ([var (program-variables prog)]) (cons var 'real))) 'real)
```

Comment:
This makes it seem like we are only going to split on reals (not posits), but we have posit code further down. Or is this there for complex numbers?